datadog/provider.go

```diff
@@ -118,19 +120,75 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 			},
 		},
 	)
-	config := datadog.NewConfiguration()
+	configV1 := datadogV1.NewConfiguration()
 	if apiURL := d.Get("api_url").(string); apiURL != "" {
-		if strings.Contains(apiURL, "datadoghq.eu") {
-			auth = context.WithValue(auth, datadog.ContextServerVariables, map[string]string{
+		parsedApiUrl, parseErr := url.Parse(apiURL)
+		if parseErr != nil {
+			return nil, fmt.Errorf(`invalid API Url : %v`, parseErr)
+		}
+		if parsedApiUrl.Host == "" || parsedApiUrl.Scheme == "" {
+			return nil, fmt.Errorf(`missing protocol or host : %v`, apiURL)
+		}
+		// Set site to "datadoghq.eu" on ServerIndex{0} if Datadog EU API URL is passed.
+		// If custom api url (not datadoghq.eu or datadoghq.com) is passed, set the api name and protocol on ServerIndex{-1}
+		if parsedApiUrl.Host == "datadoghq.eu" {
+			authV1 = context.WithValue(authV1, datadogV1.ContextServerVariables, map[string]string{
+				"site": "datadoghq.eu",
+			})
```

Comment:
I don't think we need this case. The second case covers everything, even Datadog URLs

Reply:
Updated and removed the unnecessary case you mentioned above.